### PR TITLE
[cleanup] Remove artefacts from old py34 support

### PR DESCRIPTION
**Pull request by Djailla**
## Thanks for contributing a pull request!

Remove mention of no longer supported pyhon 3.4

Following #1525 

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [N/a] added/updated test(s)
- [x] updated/extended the documentation
- [ ] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [ ] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`,`breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with "by :user:`<your username>`"
  * please use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files - by :user:`superuser`."
  * also see [examples](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

File: docs/example/basic.rst [hunk 1]
```diff
@@ -359,14 +359,14 @@ use :ref:`generative-envlist` and :ref:`conditional settings <factors>` to expre
 .. code-block:: ini
 
     [tox]
-    envlist = py{27,34,36}-django{15,16}-{sqlite,mysql}
+    envlist = py{27,36,37}-django{22,30}-{sqlite,mysql}
```
Comment (gaborbernat) [inline docs/example/basic.rst:362]: Can you switch to 39 and 38? 😄

File: docs/install.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ tox installation
 Install info in a nutshell
 ----------------------------------
 
-**Pythons**: CPython 2.7 and 3.4 or later, Jython-2.5.1, pypy-1.9ff
+**Pythons**: CPython 2.7 and 3.5 or later, Jython-2.5.1, pypy-1.9ff
```
Comment (gaborbernat) [inline docs/install.rst:7]: We still support 3.4, so this needs to go back.

File: docs/index.rst [hunk 3]
```diff
@@ -139,7 +139,7 @@ Current features
 * uses pip_ and setuptools_ by default.  Support for configuring the installer command
   through :conf:`install_command=ARGV`.
 
-* **cross-Python compatible**: CPython-2.7, 3.4 and higher, Jython and pypy_.
+* **cross-Python compatible**: CPython-2.7, 3.5 and higher, Jython and pypy_.
```
Comment (gaborbernat) [inline docs/index.rst:142]: We still support 3.4, so this needs to go back.

File: setup.cfg [hunk 4]
```diff
@@ -62,12 +62,12 @@ testing =
     flaky>=3.4.0
     freezegun>=0.3.11
     pathlib2>=2.3.3
-    pytest>=4.0.0
+    psutil>=5.6.1
     pytest-cov>=2.5.1
     pytest-mock>=1.10.0
     pytest-randomly>=1.0.0
     pytest-xdist>=1.22.2
-    psutil>=5.6.1;python_version != "3.4"
```
Comment (gaborbernat) [inline setup.cfg:70]: this still is needed for 3.4

File: tox.ini [hunk 5]
```diff
@@ -1,6 +1,5 @@
 [tox]
 envlist = py27,
-          py34,
```
Comment (gaborbernat) [inline tox.ini:3]: keep this please, we did not drop supporting this.

Comment (Djailla) [reply tox.ini:3]: @gaborbernat are you sure about py34 support ?

Here is an extract of `setup.cfg` 

`python_requires = >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*`

Comment (gaborbernat) [reply tox.ini:3]: Damn 😆

Comment (Djailla) [reply tox.ini:3]: ;)

File: docs/example/basic.rst (hunk 1 above)
Comment (Djailla) [reply docs/example/basic.rst:362]: There is no official support of Django for py39

Comment (gaborbernat) [reply docs/example/basic.rst:362]: Then let's go with 38, and 37. Instead of 36.

**Comment by Djailla** [comment]
@gaborbernat on a side question, do you consider dropping support of py27 soon ?

**Comment by gaborbernat** [comment]
> @gaborbernat on a side question, do you consider dropping support of py27 soon ?

tox 3 will probably never drop it, tox 4 will never support it.

**Pull request merged**